explore/Classifier_results-different_genes.py

```diff
@@ -0,0 +1,252 @@
+    '4089':'SMAD4',
+    '4087':'SMAD2',
+    '4221':'MEN1',
+    '4763':'NF1',
+    '4771':'NF2',
+    '7157':'TP53', 
+    '5728':'PTEN', 
+    '5925':'Rb',
+    '7428':'VHL',
+    '7486':'WRN',
+    '7490':'WT1',
+}
+
+list_of_genes = list(genes_LungCancer.keys()) + list(genes_TumorSuppressors.keys())
+
+gene_symbols = list(genes_LungCancer.values()) + list(genes_TumorSuppressors.values())
+
+AUROCS_train = []
+AUROCS_test = []
+p = []
+
+
+# In[7]:
+
+#functions
+
+def fs_mad(x, y):
+    """
+    Get the median absolute deviation (MAD) for each column of x
+    """
+    scores = mad(x) 
+    return scores, np.array([np.NaN]*len(scores))
+
+def get_threshold_metrics(y_true, y_pred):
+    roc_columns = ['fpr', 'tpr', 'threshold']
+    roc_items = zip(roc_columns, roc_curve(y_true, y_pred))
+    roc_df = pd.DataFrame.from_items(roc_items)
+    auroc = roc_auc_score(y_true, y_pred)
+    return {'auroc': auroc, 'roc_df': roc_df}
+
+
+# In[8]:
+
+for GENE in list_of_genes:
+    
+    y = Y[GENE]
+    # Here are the percentage of tumors with NF1
+    per = y.value_counts(True)
+    # Set aside 10% of the data for testing, this can only be done where the number of mutations is large enough
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=0)
+    'Size: {:,} features, {:,} training samples, {:,} testing samples'.format(len(X.columns), len(X_train), len(X_test))
+
+    # select the top features with the highest MAD
+    feature_select = SelectKBest(fs_mad, k=n_feature_kept)
+    # Include loss='log' in param_grid doesn't work with pipeline somehow
+    clf = SGDClassifier(random_state=0, class_weight='balanced',
+                    loss=param_fixed['loss'], penalty=param_fixed['penalty'])
+
+    # joblib is used to cross-validate in parallel by setting `n_jobs=-1` in GridSearchCV
+    # Supress joblib warning. See https://github.com/scikit-learn/scikit-learn/issues/6370
+    warnings.filterwarnings('ignore', message='Changing the shape of non-C contiguous array')
+    clf_grid = grid_search.GridSearchCV(estimator=clf, param_grid=param_grid, n_jobs=-1, scoring='roc_auc')
+    pipeline = make_pipeline(
+        feature_select,  # Feature selection
+        StandardScaler(),  # Feature scaling
+        clf_grid)
+    # Fit the model (the computationally intensive part)
+    pipeline.fit(X=X_train, y=y_train)
+    best_clf = clf_grid.best_estimator_
+    feature_mask = feature_select.get_support()  # Get a boolean array indicating the selected features
+    y_pred_train = pipeline.decision_function(X_train)
+    y_pred_test = pipeline.decision_function(X_test)
+
+    metrics_train = get_threshold_metrics(y_train, y_pred_train)
+    metrics_test = get_threshold_metrics(y_test, y_pred_test)
+    
+    AUROCS_train.append(metrics_train['auroc'])
+    AUROCS_test.append(metrics_test['auroc'])
+    p.append(per[1])
+  
+
+
+# In[9]:
+
+AUROCS_train_lung = AUROCS_train[0:len(list(genes_LungCancer.values()))]

```

Comment:
no need to convert `genes_LungCancer` to a list, this should work fine:

```python
AUROCS_train_lung = AUROCS_train[0:len(genes_LungCancer)]
```
This change would propagate through lines below as well